src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -332,9 +323,32 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
             out var requestId,
             out var traceState);
 
+        var activity = _activitySource.CreateActivity(
+            ActivityName,
+            ActivityKind.Server,
+            string.IsNullOrEmpty(requestId) ? null! : requestId
```

Comment:
nit: It might be a good opportunity to rename requestId variables to traceParent..